src/Fhir.Anonymizer.Core/Extensions/ElementNodeExtension.cs

```diff
@@ -47,6 +47,11 @@ public static bool IsPostalCodeNode(this ElementNode node)
             return node != null && string.Equals(node.Name, Constants.PostalCodeNodeName, StringComparison.InvariantCultureIgnoreCase);
         }
 
+        public static bool IsReferenceNode(this ElementNode node)
+        {
+            return node != null && string.Equals(node.Name, Constants.ReferenceNodeName, StringComparison.InvariantCultureIgnoreCase);
```

Comment:
Is that a reference type? Could we use type to compare?

Reply:
It might be confusing here as the target node is "Reference.reference" which is a string type. I have changed the compare rule to node.Parent is Reference type and node.name = reference. Thanks.

Reply:
Is that possible parent node is null?

Reply:
No, it's not possible. Only root node has null parent.

Reply:
Looks like if someone run this code at root node, it would throw null ref exception?

Reply:
We can change to something like:
node?.Parent?.IsReferenceNode() ?? false

Reply:
In cases of null object calling extension methods, it will not throw and pass a null value to the function.